### PR TITLE
[openrewrite] add `SanityCheck` featuring `tech.picnic.errorprone.refasterrules`

### PR DESCRIPTION
**Pull request by Pankraz76**
I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit-framework/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Method [preconditions](https://docs.junit.org/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [ ] [Coding conventions](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [ ] Change is covered by [automated tests](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [ ] Public API has [Javadoc](https://github.com/junit-team/junit-framework/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [ ] Change is documented in the [User Guide](https://docs.junit.org/snapshot/user-guide/) and [Release Notes](https://docs.junit.org/snapshot/user-guide/#release-notes)

relates to:

- https://github.com/junit-team/junit-framework/issues/5002
- https://github.com/diffplug/spotless/pull/2651

```
> Task :rewriteRun
Changes have been made to junit-platform-engine/src/main/java/org/junit/platform/engine/UniqueId.java by:
    tech.picnic.errorprone.refasterrules.CollectionRulesRecipes
        tech.picnic.errorprone.refasterrules.CollectionRulesRecipes$OptionalFirstCollectionElementRecipe
Changes have been made to junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/DefaultParallelExecutionConfigurationStrategy.java by:
    tech.picnic.errorprone.refasterrules.BigDecimalRulesRecipes
        tech.picnic.errorprone.refasterrules.BigDecimalRulesRecipes$BigDecimalSignumIsPositiveRecipe
Changes have been made to junit-platform-launcher/src/main/java/org/junit/platform/launcher/EngineFilter.java by:
    tech.picnic.errorprone.refasterrules.CollectionRulesRecipes
        tech.picnic.errorprone.refasterrules.CollectionRulesRecipes$CollectionContainsRecipe
Changes have been made to junit-platform-launcher/src/main/java/org/junit/platform/launcher/listeners/OutputDir.java by:
    tech.picnic.errorprone.refasterrules.StreamRulesRecipes
        tech.picnic.errorprone.refasterrules.StreamRulesRecipes$StreamFindAnyIsPresentRecipe
Changes have been made to jupiter-tests/src/test/java/org/junit/jupiter/engine/discovery/predicates/IsTestFactoryMethodTests.java by:
    tech.picnic.errorprone.refasterrules.StreamRulesRecipes
        tech.picnic.errorprone.refasterrules.StreamRulesRecipes$EmptyStreamRecipe
Changes have been made to platform-tests/src/test/java/org/junit/platform/commons/support/conversion/ConversionSupportTests.java by:
    tech.picnic.errorprone.refasterrules.TimeRulesRecipes
        tech.picnic.errorprone.refasterrules.TimeRulesRecipes$EpochInstantRecipe
Changes have been made to platform-tests/src/test/java/org/junit/platform/engine/support/descriptor/AbstractTestSourceTests.java by:
    tech.picnic.errorprone.refasterrules.ClassRulesRecipes
        tech.picnic.errorprone.refasterrules.ClassRulesRecipes$ClassIsInstanceRecipe
Changes have been made to platform-tooling-support-tests/src/test/java/platform/tooling/support/tests/OutputAttachingExtension.java by:
    tech.picnic.errorprone.refasterrules.ComparatorRulesRecipes
        tech.picnic.errorprone.refasterrules.ComparatorRulesRecipes$ReverseOrderRecipe
Changes have been made to platform-tooling-support-tests/src/test/java/platform/tooling/support/tests/ModularUserGuideTests.java by:
    tech.picnic.errorprone.refasterrules.StreamRulesRecipes
        tech.picnic.errorprone.refasterrules.StreamRulesRecipes$StreamFilterSortedRecipe
Changes have been made to platform-tooling-support-tests/src/test/java/platform/tooling/support/tests/LocalMavenRepo.java by:
    tech.picnic.errorprone.refasterrules.ComparatorRulesRecipes
        tech.picnic.errorprone.refasterrules.ComparatorRulesRecipes$ReverseOrderRecipe
Please review and commit the results.
Estimate time saved: 50m
```

### CODE REVIEW

File: gradle/rewrite.gradle [hunk 1]
```diff
@@ -0,0 +1,17 @@
+apply plugin: "org.openrewrite.rewrite"
+
+rewrite {
+	activeRecipe("org.junit.jupiter.openrewrite.SanityCheck")
+	exclusions.addAll("**.ci-temp**")
```
Comment (Pankraz76) [inline gradle/rewrite.gradle]:
```suggestion
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -49,6 +49,11 @@ jobs:
       uses: codecov/codecov-action@5a1091511ad55cbe89839c7260b706298ca349f7 # v5.5.1
       with:
         token: ${{ secrets.CODECOV_TOKEN }}
+    - name: rewriteDryRun
```
Comment (Pankraz76) [inline .github/workflows/main.yml:52]: this is just an reminder of course you need to choose where to put this best. 

- https://github.com/diffplug/spotless/blob/main/.github/workflows/ci.yml#L34

File: rewrite.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  - tech.picnic.errorprone.refasterrules.CollectionRulesRecipes
+  - tech.picnic.errorprone.refasterrules.ComparatorRulesRecipes
+  - tech.picnic.errorprone.refasterrules.FileRulesRecipes
+  - tech.picnic.errorprone.refasterrules.MicrometerRulesRecipes
+  - tech.picnic.errorprone.refasterrules.PatternRulesRecipes
+  - tech.picnic.errorprone.refasterrules.PreconditionsRulesRecipes
+  - tech.picnic.errorprone.refasterrules.PrimitiveRulesRecipes
+  - tech.picnic.errorprone.refasterrules.StreamRulesRecipes
+  - tech.picnic.errorprone.refasterrules.TimeRulesRecipes
+  # - org.openrewrite.staticanalysis.EqualsAvoidsNull
```
Comment (Pankraz76) [inline rewrite.yml:27]: this does change alot despite the initial effort. Every tool has its benefits making it win to the community.

**Review by Pankraz76 — COMMENTED** [review]
kindly request feedback. Thanks.

**Comment by marcphilipp** [comment]
We have already decided against adopting OpenRewrite for our day-to-day build. I'd be open to looking at Error Prone's refaster with Picnic's rules, though, as discussed on the issue.
